yb-voyager/src/srcdb/postgres.go

```diff
@@ -507,3 +509,31 @@ func (pg *PostgreSQL) ClearMigrationState(migrationUUID uuid.UUID, exportDir str
 	log.Infof("ClearMigrationState not implemented yet for PostgreSQL")
 	return nil
 }
+
+func (pg *PostgreSQL) GetReplicationConnection() (*pgconn.PgConn, error) {
+	return pgconn.Connect(context.Background(), pg.getConnectionUri()+"&replication=database")
+}
+
+func (pg *PostgreSQL) CreateLogicalReplicationSlot(conn *pgconn.PgConn, migrationUUID uuid.UUID, dropIfAlreadyExists bool) (pglogrepl.CreateReplicationSlotResult, error) {
+	replicationSlotName := fmt.Sprintf("voyager_%s", strings.Replace(migrationUUID.String(), "-", "_", -1))
+	res, err := pglogrepl.CreateReplicationSlot(context.Background(), conn, replicationSlotName, "pgoutput",
+		pglogrepl.CreateReplicationSlotOptions{Mode: pglogrepl.LogicalReplication})
+
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") && dropIfAlreadyExists {
+			// drop and recreate
+			err = pglogrepl.DropReplicationSlot(context.Background(), conn, replicationSlotName, pglogrepl.DropReplicationSlotOptions{})
+			if err != nil {
+				return pglogrepl.CreateReplicationSlotResult{}, fmt.Errorf("failed to delete existing replication slot: %v", err)
+			}
+			res, err = pglogrepl.CreateReplicationSlot(context.Background(), conn, replicationSlotName, "pgoutput",
+				pglogrepl.CreateReplicationSlotOptions{Mode: pglogrepl.LogicalReplication})
+			if err != nil {
+				return pglogrepl.CreateReplicationSlotResult{}, fmt.Errorf("failed to create replication slot: %v", err)
```

Comment:
In the error wrapper, it is better to just record the **action** the code was trying to perform (as opposed to `"failed to perform an action"`). e.g. This example can be written as:
```
fmt.Errorf("create replication slot %q: %v", slotName, err)
```
When wrappers messages are written like this the final message looks like: `action1: action2: action3: err` and the go authors call this as "semantic stack trace". An example of final error message: `init app: load config: open /etc/app.json: file not found`. Compare it with `failed to init app: failed to load config: failed to open /etc/app.json: file not found`.